### ISSUE TITLE
[Component Mode] Entity Picker leads to a crash when used in Component Mode

### ISSUE THREAD

**Issue by mbalfour-amzn** [o3de/o3de#14029]
**Describe the bug**
While in Component Mode, if the component being edited has an Entity Picker field accessible on it, attempting to use the Entity Picker will lead to unpredictable behavior and ultimately a crash.

**Steps to reproduce**
1. Open the Editor with a new or existing level.
2. Create a new entity with an Image Gradient, a Gradient Transform Modifier, and a Box Shape.
3. On the Image Gradient, select any available existing image for the Image Asset.
4. On the Image Gradient, press the Edit button to start the paintbrush component mode.
5. On the Image Gradient, press the Entity Picker icon next to the "Pin Preview to Entity" field.
6. Press Esc to exit entity picking.

**Expected behavior**
The user is able to pick an entity, despite being in component mode.

**Actual behavior**
After step 5, the component mode partially ends, though the Entity Outliner remains greyed out and the "Edit" button still says "Done" but does nothing when pressed.

After step 6, the Editor crashes.

The callstack of the crash is as follows:
```
 	Qt5Core.dll!00007ff837f3a279()	Unknown
>	[Inline Frame] EditorLib.dll!QObject::findChildren(const QString &) Line 184	C++
 	EditorLib.dll!ShortcutDispatcher::shortcutFilter(QObject * obj, QShortcutEvent * shortcutEvent) Line 394	C++
 	[External Code]	
 	EditorLib.dll!CryEditMain(int argc, char * * argv) Line 3997	C++
 	[Inline Frame] Editor.exe!AZStd::Internal::INVOKE(int(*)(int, char * *) &) Line 208	C++
 	[Inline Frame] Editor.exe!AZStd::invoke(int(*)(int, char * *) &) Line 54	C++
 	Editor.exe!main(int argc, char * * argv) Line 32	C++
 	Editor.exe!WinMain(HINSTANCE__ * __formal, HINSTANCE__ * __formal, char * __formal, int __formal) Line 97	C++
 	[External Code]	
```

**Screenshots/Video**
https://user-images.githubusercontent.com/82224783/212152251-a2ae1463-49cc-4b5e-aee7-a72e6f37a1e8.mp4

**Found in Branch**
development

**Commit ID from [o3de/o3de](https://github.com/o3de/o3de) Repository**
5779140a90977e46e33a1db51ff241652294e341

**Additional context**
I'm currently trying out potential solutions to this problem.

**Comment by hultonha** [comment]
Thanks for writing this up @mbalfour-amzn 👍

**Comment by mbalfour-amzn** [comment]
This is now fixed, with #14032 , but there are still bugs with pressing the Esc key when using the new Action Manager (see #13467 and #14146 ).

**Comment by hultonha** [comment]
Awesome to hear, thanks @mbalfour-amzn! 🥳
